examples/plot_anomaly_comparison.py

```diff
@@ -0,0 +1,120 @@
+"""
+========================================================================
+Comparing different anomaly/novelty detection algorithms on toy datasets
+========================================================================
+
+This example shows characteristics of different
+anomaly/novelty detection algorithms on 2D datasets. Datasets contain
+one or two modes (regions of high density) to illustrate the ability
+of algorithms to cope with multimodal data.
+
+While these examples give some intuition about the
+algorithms, this intuition might not apply to very high
+dimensional data.
+"""
+print(__doc__)  # noqa
+
+import time
+
+import numpy as np
+from scipy import stats
+import matplotlib
+import matplotlib.pyplot as plt
+
+from sklearn import svm
+from sklearn.datasets import make_moons
+from sklearn.covariance import EllipticEnvelope
+from sklearn.ensemble import IsolationForest
+from sklearn.neighbors import LocalOutlierFactor
+
+matplotlib.rcParams['contour.negative_linestyle'] = 'solid'
+
+# Example settings
+n_samples = 300
+outliers_fraction = 0.15
+clusters_separation = (0, 1, 2)
+n_outliers = int(outliers_fraction * n_samples)
+n_inliers = n_samples - n_outliers
+
+
+def make_dataset(n_inliers, offset, stds=(0.5, 0.5), random_state=0):
+    rng = np.random.RandomState(random_state)
+    # Data generation
+    X1 = stds[0] * rng.randn(n_inliers // 2, 2) - offset
+    X2 = stds[1] * rng.randn(n_inliers // 2, 2) + offset
+    X = np.concatenate([X1, X2], axis=0)
+    return X
+
+# define outlier/anomaly detection methods to be compared
+anomaly_algorithms = [
+    ("Robust covariance", EllipticEnvelope(contamination=outliers_fraction)),
+    ("One-Class SVM", svm.OneClassSVM(nu=0.95 * outliers_fraction + 0.05,
+                                      kernel="rbf", gamma=0.1)),
+    ("Isolation Forest", IsolationForest(contamination=outliers_fraction,
+                                         random_state=42)),
+    ("Local Outlier Factor", LocalOutlierFactor(
+        n_neighbors=35, contamination=outliers_fraction))]
+
+# Define datasets
+datasets = [
+    make_dataset(n_inliers, stds=(0.5, 0.5), offset=0.),
+    make_dataset(n_inliers, stds=(0.3, 1.5), offset=2.),
+    4. * (make_moons(n_samples=n_samples, noise=.05)[0] -
```

Comment:
you do the multiplication so the meshgrid is the same for each?

Reply:
Oh and I guess so that the outliers can be on the same scale? (maybe add a comment?)